// @copilotkit/runtime/dist/graphql/types/converted/index.mjs
import "reflect-metadata";
import { MessageRole } from "../enums.mjs";
import { MessageStatusCode } from "../message-status.type.mjs";
import { randomId } from "@copilotkit/shared";

//#region src/graphql/types/converted/index.ts
var Message = class {
	constructor(props) {
		props.id ??= randomId();
		props.status ??= { code: MessageStatusCode.Success };
		props.createdAt ??= /* @__PURE__ */ new Date();
		Object.assign(this, props);
	}
	isTextMessage() {
		return this.type === "TextMessage";
	}
	isActionExecutionMessage() {
		return this.type === "ActionExecutionMessage";
	}
	isResultMessage() {
		return this.type === "ResultMessage";
	}
	isAgentStateMessage() {
		return this.type === "AgentStateMessage";
	}
	isImageMessage() {
		return this.type === "ImageMessage";
	}
};
const Role = MessageRole;
var TextMessage = class extends Message {
	constructor(props) {
		super(props);
		this.type = "TextMessage";
		this.type = "TextMessage";
	}
};
var ActionExecutionMessage = class extends Message {
	constructor(..._args) {
		super(..._args);
		this.type = "ActionExecutionMessage";
	}
};
var ResultMessage = class ResultMessage extends Message {
	constructor(..._args2) {
		super(..._args2);
		this.type = "ResultMessage";
	}
	static encodeResult(result, error) {
		const errorObj = error ? typeof error === "string" ? {
			code: "ERROR",
			message: error
		} : error instanceof Error ? {
			code: "ERROR",
			message: error.message
		} : error : void 0;
		if (errorObj) return JSON.stringify({
			error: errorObj,
			result: result || ""
		});
		if (result === void 0) return "";
		return typeof result === "string" ? result : JSON.stringify(result);
	}
	static decodeResult(result) {
		if (!result) return { result: "" };
		try {
			const parsed = JSON.parse(result);
			if (parsed && typeof parsed === "object") {
				if ("error" in parsed) return {
					error: parsed.error,
					result: parsed.result || ""
				};
				return { result: JSON.stringify(parsed) };
			}
			return { result };
		} catch (e) {
			return { result };
		}
	}
	hasError() {
		try {
			const { error } = ResultMessage.decodeResult(this.result);
			return !!error;
		} catch {
			return false;
		}
	}
	getError() {
		try {
			const { error } = ResultMessage.decodeResult(this.result);
			return error;
		} catch {
			return;
		}
	}
};
var AgentStateMessage = class extends Message {
	constructor(..._args3) {
		super(..._args3);
		this.type = "AgentStateMessage";
	}
};
var ImageMessage = class extends Message {
	constructor(..._args4) {
		super(..._args4);
		this.type = "ImageMessage";
	}
};

//#endregion
export { ActionExecutionMessage, AgentStateMessage, ImageMessage, ResultMessage, Role, TextMessage };
